### ISSUE TITLE
Can't resolve 'mml-react' in '/node_modules/stream-chat-react/dist' when y `yarn install --ignore-optional`

### ISSUE THREAD

**Issue by irodriguezagentero** [GetStream/stream-chat-react#908]
There is an error with the library when we bundle our application, in order to avoid unnecessary packages on installation we use the **yarn** option [`--ignore-optional`](https://classic.yarnpkg.com/en/docs/cli/install/#toc-yarn-install-ignore-optional), doing it makes the build fails because **stream-chat-react** uses it anyway buy it has as **optionalDependency**.

https://github.com/GetStream/stream-chat-react/blob/addef01a9f356548b0013b4281cfb74d5807958b/package.json#L50

**Comment by irodriguezagentero** [comment]
@DanC5 Be careful when you use CRA because it install all dependencies by default, if you don't clean `node_modules` and after that you run `yarn install --ignore-optional` you'll have all the dependencies. I have created a [repo](https://github.com/irodriguezagentero/stream-chat-react-issue) to let you try (in this case is done with NextJS).

What you commented about adding a **resolve** field in Webpack is impossible because the problem isn't that the resolution of the package is wrong, the problem is that the dependency is not installed (caused because it is optional and we ignore it in the install process) if i add a custom resolution i'll get the same error. As i see the dependency is marked as optional but it is used anyway in the code and that's the reason.

**Comment by petyosi** [comment]
@irodriguezagentero Indeed, to make that work needs a few more steps. The dependency is used in the code, but, given that you are not going to use MML, it will never be executed. We are yet to figure out how to make that work out of the box build time. 

In the meantime, here's what you can do, assuming that you don't want to `eject` your CRA setup:

1. Install [craco](https://github.com/gsoft-inc/craco/blob/master/packages/craco/README.md#installation).
2. Create an empty file called `mml-react-stub.js` in your root folder (you can change that up to your preferences).
3. Use the following `craco.config.js`:

```js
module.exports = {
  webpack: {
    resolve: {
      alias: {
        "mml-react": "./mml-react-stub.js",
      },
    },
  },
};
```
This will make the build pass, while not adding the additional cost of `mml-react`.

**Comment by irodriguezagentero** [comment]
@petyosi Thanks for the response. i added a fix to avoid the issue but i'll take a look to this to remove the patch as soon as i can.

**Comment by DanC5** [comment]
@irodriguezagentero between your patch and Petyo's proposed fix, it sounds like we have a solution here so I'm going to close the issue, but don't hesitate to reach back out as needed. Thanks.

**Comment by DanC5** [comment]
@irodriguezagentero one more option, we just tested this solution on your example repo and it worked. Add this to your next.config.js

```
module.exports = {
  webpack: (config, { webpack }) => {
    config.plugins.push(new webpack.IgnorePlugin(/mml-react/));
    return config;
  },
};
```
